include/networkit/matching/BSuitorMatcher.hpp

```diff
@@ -22,19 +22,21 @@ namespace NetworKit {
  * B-Suitor matching finding algorithm.
  */
 class BSuitorMatcher : public BMatcher {
-public:
+protected:
     struct MatchingNode {
         node id;
         edgeweight weight;
 
-        MatchingNode() = default;
+        MatchingNode() : id(none), weight(0) {}
         MatchingNode(node n, edgeweight w) : id(n), weight(w) {}
 
-        bool operator==(const MatchingNode &other) const {
```

Comment:
this function has to be defined explicitly (as `=default`). `!=` will the be generated implicitly.